### ISSUE TITLE
[BUG] deepseek-v4 `comb.to(dtype).transpose(-1, -2)`

### ISSUE THREAD

**Issue by Jintao-Huang** [huggingface/transformers#46172]
### System Info

Expected fix:`comb.to(dtype).transpose(-1, -2)` -> `comb.to(dtype)`

https://github.com/huggingface/transformers/blame/10555512868d663ee1ff627e4f5c5c260114235b/src/transformers/models/deepseek_v4/modular_deepseek_v4.py#L998-L999

https://huggingface.co/deepseek-ai/DeepSeek-V4-Flash/blob/main/inference/model.py#L685-L686

comb: [b,s,hc,hc], y: [b,s,hc,d]

torch.sum(comb.unsqueeze(-1) * residual.unsqueeze(-2), dim=2)

[b, s, hc, hc, 1] * [b, s, hc, 1, d] -[*]> [b, s, hc, hc, d] -[sum]-> [b, s, hc, d]

### Who can help?

-

### Information

- [ ] The official example scripts
- [ ] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

-

### Expected behavior

-

**Comment by Jintao-Huang** [comment]
Sorry, it looks like there's no issue with it.
